lib/parse-args.js

```diff
@@ -194,6 +194,52 @@ function buildYargs (withCommands = false) {
   return yargs
 }
 
+function loadConfigFile (path) {
+  let config = {}
+  const jsExts = ['.js', '.cjs']
+  const ymlExts = ['.yml', '.yaml']
+  const fileName = basename(path)
+
+  const ext = extname(path).toLowerCase()
+  if (jsExts.includes(ext)) {
+    config = require(path)
+  } else if (ymlExts.includes(ext)) {
+    config = require('js-yaml').load(readFileSync(path, 'utf8'))
+  } else if (ext === '.json' || fileName.slice(-2) === 'rc') {
+    config = JSON.parse(readFileSync(path, 'utf8'))
+  }
+
+  // Should the process die if none of these filename extensions are found?
+  if (Object.keys(config).length === 0) {
+    throw new Error(`Unsupported file type ${ext} while reading file ${path}`)
```

Comment:
I read why this was added, but it's going to result in bad UX when a config file return an empty object - in that situation it loaded fine, and is not an invalid file type, it's just not got any config entries.

Reply:
I want to be sure I am clear on what you stated above.  To rewrite what you expressed in two use cases.

Scenario:  

A user executes the c8 command with the --config parameter set to an existing file on the file system

Given:

  The file has a valid file name as documented in the c8 readme
  The file has no contents.

Scenario:  

A user executes the c8 command without additional configuration arguments and c8 discovers a configuration file.

Given:
   The file has a valid file name as documented in the c8 readme
   The file has no contents.


Can you please confirm?

The line does cover the following use case:

Scenario:
  A user executes the c8 command with the --config parameter set to an existing file on the file system.

Given:
  The file does not have a valid file name as documented in the c8 readme

Result:
   c8 will produce the error `Unsupported file type ...'

This test case can be found on [Line 70 - Line 85](https://github.com/bcoe/c8/blob/22a9c08a01a219a97e9768be7b211a6793c22956/test/parse-args.js#L70-L85) on the file test/parse-args.js

Reply:
Some notes before I hit the scenarios:

1. Only YAML would have an empty file return an empty object.
2. An empty JSON file is a syntax error.
3. A JSON file of `{}` is valid.
4. A CJS script that doesn't export anything IMO SHOULD result in an error as this is a common mistake.
5. A CJS script that exports an empty object, e.g. `{}` is valid. 

So, scenarios:

----
Scenario:
A user executes the c8 command with the --config parameter set to an existing file on the file system
OR
A user executes the c8 command without the --config parameter set AND c8 discovers a configuration file.

Given any of:
- The file is JSON with the `.json` extension, case insensitive.
  The file has the contents `{}` or equivalent.
- The file is YAML with the `.yaml` or `.yml` extension, case insensitive.
  The file has no contents
- The file is CommonJS with the `.js` or `.cjs` extension, case insensitive.
  The file's contents result in an empty object `{}` being exported.

Result:
c8 will run with a default configuration.

----
Scenario:
A user executes the c8 command with the --config parameter set to an existing file on the file system
OR
A user executes the c8 command without the --config parameter set AND c8 discovers a configuration file.

Given any of:
- The file has the `.json` extension, case insensitive.
  The file is invalid JSON.
- The file has the `.yaml` or `.yml` extension, case insensitive.
  The file is invalid YAML.
- The file has the `.js` or `.cjs` extension, case insensitive.
  The file's contents are invalid CommonJS or have a runtime error.
- The file has the `.js` or `.cjs` extension, case insensitive.
  The file doesn't export anything.
- The file has the `.js` or `.cjs` extension, case insensitive.
  The file exports something that's not an object.

Result:
c8 will fail with an appropriate error based on the actual underlying problem that will be helpful to the user.

Some errors are really very obtuse or otherwise mask the real problem. I think over time of people testing this those errors will be discovered and made nicer, so making the errors nicer probably shouldn't be part of this PR.

----
Scenario:
A user executes the c8 command with the --config parameter set to an existing file on the file system.

Given:
The file name is NOT one of the ones documented in the c8 readme.
The file extension is one of the valid types, e.g. `.js, .cjs, .json, .yaml, .yml` etc.
The file returns a valid c8 configuration.

Result:
c8 will run with the configuration coming from the file given.

This is because there are several reasons for using the `--config` parameter, here's just a few:
1. I've got multiple config files to choose from. This you've covered.
2. My config file's at an unusual path that auto detection won't find, e.g. `.configs/c8rc.yaml` This I'm pretty sure you've covered, though maybe not directly with a test.
3. I don't like any of the default file names and have my own naming scheme I use, e.g. `.config.c8.yaml` - the current code looks like it handles this case already, there just doesn't seem to be a test covering this idea.

Reply:
I've [created a PR](https://github.com/mcknasty/c8/pull/5) for the branch of this PR that adds tests and changes to support the above.